src/main/java/uk/gov/ida/matchingserviceadapter/rest/MatchingServiceRequestDto.java

```diff
@@ -36,6 +38,20 @@ public MatchingServiceRequestDto(
         this.levelOfAssurance = levelOfAssurance;
     }
 
+    public MatchingServiceRequestDto(
+        EidasMatchingDataset eidasDataset,
+        Optional<Cycle3DatasetDto> cycle3Dataset,
+        String hashedPid,
+        String matchId,
+        LevelOfAssuranceDto levelOfAssurance) {
+
+        this.eidasDataset = eidasDataset;
+        this.cycle3Dataset = cycle3Dataset;
+        this.hashedPid = hashedPid;
+        this.matchId = matchId;
+        this.levelOfAssurance = levelOfAssurance;
+    }
```

Comment:
There should be a common private ctor that by default init the hashedPid, matchId and Loa so the two used by calling ```this(hashedPid, matchId, levelOfAssurance);```